ddcommon-ffi/src/endpoint.rs

```diff
@@ -10,9 +10,12 @@ use std::str::FromStr;
 #[no_mangle]
 #[must_use]
 pub extern "C" fn ddog_endpoint_from_url(url: crate::CharSlice) -> Option<Box<Endpoint>> {
-    parse_uri(url.to_utf8_lossy().as_ref())
-        .ok()
-        .map(|url| Box::new(Endpoint { url, api_key: None }))
+    parse_uri(url.to_utf8_lossy().as_ref()).ok().map(|url| {
+        Box::new(Endpoint {
+            url,
+            ..Default::default()
+        })
```

Comment:
This pattern happens enough we should just have a constructor `Endpoint::from_url()`